internal/exec/stages/disks/luks.go

```diff
@@ -247,13 +256,37 @@ func (s *stage) createLuks(config types.Config) error {
 
 		args = append(args, devAlias)
 
+		// append the zkey specific luksFormat arguments
+		if luks.Cex.IsPresent() {
+			err := s.zkeySecKeyGen(luks)
+			if err != nil {
+				return fmt.Errorf("zkey: secure key generation failed: %w", err)
+			}
+			// Append the zkey cryptsetup specific parameter for luksFormat
+			cex_args, err := s.zkeySecCryptGenArgs(luks)
+			if err != nil {
+				return fmt.Errorf("Error generating key size %w", err)
+			}
+			args = append(args, cex_args...)
+		}
+
 		if _, err := s.Logger.LogCmd(
 			exec.Command(distro.CryptsetupCmd(), args...),
 			"creating %q", luks.Name,
 		); err != nil {
 			return fmt.Errorf("cryptsetup failed: %v", err)
 		}
 
+		if luks.Cex.IsPresent() {
+			err := s.zkeyCryptSetvp(luks, keyFilePath)
+			if err != nil {
+				return fmt.Errorf("zkey cryptsetup setvp failed: %w", err)
```

Comment:
I'd just `return err` here since the error context within `s.zkeyCryptSetvp()` seems sufficient.

Reply:
Done